### PR TITLE
Add support for binary operation sub, mul, or, and, xor, equal, less,…

### PR DESCRIPTION
**Pull request by tianboh**
Add support for part of binary operations

### CODE REVIEW

File: python/halo/halo.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 debug = os.environ.get("CANAL_DEBUG")
 debug = debug is not None and debug != "0"
 
-LIB_HALO = "libhalo.so"
+LIB_HALO = "/host/heterogeneity-aware-lowering-and-optimization/build/lib/libhalo.so"
```
Comment (weimingzha0) [inline python/halo/halo.py:24]: remove this?

Comment (tianboh) [reply python/halo/halo.py:24]: I am modifying this...

**Review by weimingzha0 — APPROVED** [review]

File: ODLA/platforms/dnnl/odla_dnnl.h [hunk 2]
```diff
@@ -280,4 +280,13 @@ static inline odla_value CreateValue(const dnnl::memory& mem,
   return ret;
 }
 
+static dnnl::memory broadcast_mem(const dnnl::memory orig_mem,
```
Comment (weimingzha0) [inline ODLA/platforms/dnnl/odla_dnnl.h:283]: static declaration won't work for odla_dnnl_binary.cc

Comment (weimingzha0) [reply ODLA/platforms/dnnl/odla_dnnl.h:283]: you can either move the definition here (like line 273) or follow the example in line 36.

File: include/halo/lib/ir/values.h [hunk 3]
```diff
@@ -313,8 +313,8 @@ class IRObject {
   }
 
   /// Return the attributes list.
-  const std::vector<std::unique_ptr<Attribute>>& GetAttributes() const
-      noexcept {
+  const std::vector<std::unique_ptr<Attribute>>& GetAttributes()
```
Comment (weimingzha0) [inline include/halo/lib/ir/values.h:316]: why change here?

Comment (tianboh) [reply include/halo/lib/ir/values.h:316]: its automatically modified...

Comment (tianboh) [reply include/halo/lib/ir/values.h]: changed to previous one.

**Pull request merged**